corehq/apps/repeaters/models.py

```diff
@@ -313,31 +315,13 @@ def handle_response(self, result, repeat_record):
         """
         if isinstance(result, Exception):
             repeat_record.handle_exception(result)
-            self.handle_exception(result, repeat_record)
+            self.generator.handle_exception(result, repeat_record)
```

Comment:
Yeah, I think it's not super clear where the right place is to put these callbacks, but on the repeater kind of makes more sense to me too. While we're on the topic, this thing of linking repeater payload generators to repeaters through a decorator is totally insane. (I probably wrote it). It'd make a lot more sense for the repeater to have `payload_generators` dict that mapped formats to payload generators.

Anyway, I think all of that can wait for another PR, but I think that's a good suggestion.